ENIGMAsystem/SHELL/Graphics_Systems/OpenGL3/GL3textures.cpp

```diff
@@ -47,10 +47,10 @@ vector<TextureStruct*> textureStructs(0);
       show_error("Attempting to access non-existing texture " + toString(texid), false);\
       return v;\
     }\
-    const int tex = (texid==-1?-1:textureStructs[texid]->gltex);
+    const unsigned tex = (texid==-1?0:textureStructs[texid]->gltex);
```

Comment:
Related to the above. Are we quite certain that zero means none? Nothing seems to constrain that `GL_TEXTURE0` isn't actually 0, and `GL_TEXTURE0` is a valid texture.

Reply:
Yes, I have read that `0` is the "default texture object" according to the OpenGL standard.
https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_sampler_objects.txt

>Textures in GL are represented by named objects. The name space for
    texture objects is the unsigned integers, with zero reserved by the
    GL to represent the default texture object. The default texture
    object is bound to each of the TEXTURE_1D, TEXTURE_2D, TEXTURE_3D,
    TEXTURE_1D_ARRAY, TEXTURE_2D_ARRAY, TEXTURE_RECTANGLE,
    TEXTURE_BUFFER, TEXTURE_CUBE_MAP, TEXTURE_2D_MULTISAMPLE, and
    TEXTURE_2D_MULTISAMPLE_ARRAY targets during context initialization.